File: Dev/unitTest_Engine_cs/Graphics/2D/PlayEffectImmediately.cs
using asd;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace unitTest_Engine_cs.Graphics._2D
{
	class PlayEffectImmediately : EngineTest
	{
		public PlayEffectImmediately()
			: base(60)
		{
		}

		protected override void OnStart()
		{
			var effect = Engine.Graphics.CreateEffect("Data/Effect/Laser01.efk");
			var effectObj = new asd.EffectObject2D();

			effectObj.Position = new asd.Vector2DF(320, 240);
			effectObj.Scale = new asd.Vector2DF(20, 20);
			effectObj.EffectRotation = 90;
			effectObj.Effect = effect;

			Engine.AddObject2D(effectObj);
			effectObj.Play();
		}
	}
}

File: Dev/unitTest_Engine_cpp_gtest/Graphics/2D/PlayEffectImmediately.cpp
#include "../../EngineTest.h"
#include <Altseed.h>
#include <gtest/gtest.h>
#include <memory>

using namespace std;
using namespace asd;

class Graphics_PlayEffectImmediately : public EngineTest
{
public:
	Graphics_PlayEffectImmediately(bool isOpenGLMode) : EngineTest(u"PlayEffectImmediately", isOpenGLMode, 60) {}

protected:
	void OnStart()
	{
		auto effect = Engine::GetGraphics()->CreateEffect(u"Data/Effect/Laser01.efk");
		auto effectObj = make_shared<EffectObject2D>();

		effectObj->SetPosition(asd::Vector2DF(320, 240));
		effectObj->SetScale(asd::Vector2DF(20, 20));
		effectObj->SetEffectRotation(90);
		effectObj->SetEffect(effect);

		Engine::AddObject2D(effectObj);

		effectObj->Play();
	}
};

ENGINE_TEST(Graphics, PlayEffectImmediately)